### PR TITLE
Tidy deps

### PR DESCRIPTION
**Pull request by linusnorton**
Remove as many dependencies as possible. 

Enable linting.

Fix some issues because of linting.

### CODE REVIEW

File: api/lib/interceptors.ts [hunk 1]
```diff
@@ -1,43 +1,43 @@
...
-import * as log4js from 'log4js'
-import { config } from '../../config'
-import { shorten, valueOrNull } from '../lib/util'
+import * as exceptionFormatter from 'exception-formatter';
+import * as log4js from 'log4js';
+import { config } from '../../config';
+import { shorten, valueOrNull } from '../lib/util';
 
 const exceptionOptions = {
     maxLines: 1,
```
Comment (velimd) [inline api/lib/interceptors.ts]: question: is the comma needed, is it best practice

File: api/lib/request/mockRequest.ts [hunk 2]
```diff
@@ -1,5 +1,4 @@
 const request = require('request-promise')
```
Comment (velimd) [inline api/lib/request/mockRequest.ts]: why doesnt this need comma? if others require it

File: api/lib/interceptors.ts (hunk 1 above)
Comment (linusnorton) [reply api/lib/interceptors.ts]: it is neither, some people prefer it though. I generally don't leave trailing commas

File: api/lib/request/mockRequest.ts (hunk 2 above)
Comment (linusnorton) [reply api/lib/request/mockRequest.ts]: will discuss over slack, essentially the person that wrote this was a JS heretic.

**Pull request merged**